File: leetcode/2024/python/1/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i, j in enumerate(nums):
            s = target - j
            if s in m:
                return [m[s], i]
            m[j] = i
        return []

File: leetcode/2024/cpp/1/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> s;
        int complement = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            complement = target - nums[i];
            if (s.find(complement) != s.end())
            {
                return {s[complement], i};
            }
            s[nums[i]] = i;
        }
        return {};
    }
};
